xml_analyzer/xmlreader.py

```diff
@@ -1,144 +1,153 @@
-                        ref_constr_block_id = self.root.find(
+                        ref_constr_block_id = self._root.find(
                             './/*[@Id="{}"]/../..'.format(ref_constr_param_id)
                         ).get('Id')
-                        ref_constr_prop_id = self.root.find(
+                        ref_constr_prop_id = self._root.find(
                             './/SysMLConstraintProperty//*[@Idref="{}"]/../..'.format(ref_constr_block_id)
                         ).get('Id')
-                        property_name = self.root.find('.//*[@Id="{}"]'.format(binding_connector.get('To'))).get('Name')
+                        property_name = self._root.find('.//*[@Id="{}"]'.
+                                                        format(binding_connector.get('To'))).get('Name')
                         dependencies.append(dependency(property_name, ref_constr_prop_id))
 
         return dependencies
 
-    def find_constraint_spec(self, constraint_property: Et.Element) -> str:
+    def _find_constraint_spec(self, constraint_property: Et.Element) -> str:
         query = ".//ConstraintElement/ConstrainedElements//*[@Idref='{}']/../..//CompositeValueSpecification"\
             .format(constraint_property.get("Id"))
-        constraint_spec = self.root.find(query).get('Value')
+        constraint_spec = self._root.find(query).get('Value')
         return constraint_spec
 
-    def build_data_container(self, constraint_property: Et.Element) -> DataContainer:
-
-        calculation_data = DataContainer()
-
-        binding_connectors = self.resolve_bindings(constraint_property)
-        dependencies = self.resolve_dependencies(binding_connectors)
-
-        constraint_spec = self.find_constraint_spec(constraint_property)
-        autocalc_sysml_type_names = self.parse_constraint_spec(constraint_spec)
-        calculation_data.set_constraint_specification(constraint_spec)
+    def _find_constraint_property(self, val: str, method='by_id'):
+        element = None
+        if method == 'by_id':
+            query = ".//SysMLConstraintProperty[@Id='{}']".format(val)
+            element = self._root.find(query)
 
-        calculation_data.add_dependencies(dependencies)
+        return element
 
+    def _find_initial_values(self, binding_connectors: List, calculation_data: DataContainer):
         for binding_connector in binding_connectors:
             id_from = binding_connector.get('From')
             id_to = binding_connector.get('To')
-            val = self.root.find(".//*[@Id='{}']".format(id_from)).get('InitialValue')
-            prop = self.root.find(".//*[@Id='{}']".format(id_to)).get('Name')
+            val = self._root.find(".//*[@Id='{}']".format(id_from)).get('InitialValue')
+            prop = self._root.find(".//*[@Id='{}']".format(id_to)).get('Name')
             stereo = binding_connector.find('./Stereotypes/Stereotype[@Name="external"]')
             if stereo is not None:
```

Comment:
could be simplified

Reply:
Fixed with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/ffeb519bb77b6dd72bd2e24167623a3a70184ea1

Reply:
Turn out this does not work. Reverted to former state with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/5e9632342512a874b9746546b01a3087305c6dd9